.changeset/purple-flies-end.md

```diff
@@ -0,0 +1,13 @@
+---
+"@razorpay/blade": minor
+---
+
+feat: Added internationalization in Amount component via i18nify.
+References
+- **i18nify-js:** https://www.npmjs.com/package/@razorpay/i18nify-js
+- **i18nify-react:** https://www.npmjs.com/package/@razorpay/i18nify-react
+
+How to update ? 
+1. Install i18nify peer dependency `yarn add @razorpay/i18nify-js`
+2. _[Optional]_: Install i18nify-react peer dependency to manage state effectively `yarn add @razorpay/i18nify-react`
+3. Install latest Blade `yarn add @razorpay/blade@latest`
```

Comment:
Should we ask consumers to wrap their apps with I18nProvider here? otherwise all their existing Amount components will start getting formatted as their browser config no?

Reply:
We should add a note about this

Reply:
IMO, existing Amount components should default back to browser settings only. 
But, I've added a brief context and guide in the changelog for developers who wish to maintain a fixed locale or if they provide users the option to change page locale. 

We should probably talk about `i18nify` and `i18nify-react` in installation steps in the Blade documnetation as well. 

_Note -> JFYI, `I18nProvider` (i18nify-react) is an optional feature. Nothing would change if developers wrap their application inside this provider but do not set any state._

Reply:
> IMO, existing Amount components should default back to browser settings only.

The issue is the existing Amount components. For example, if someone is displaying `Rs 10L` right now. It will turn into `Rs 1M` without a breaking change. And Rs 10L -> Rs 1M is a big enough change for product and design.

Internally also we checked in team and most people have en-US set on their browser

![image](https://github.com/razorpay/blade/assets/30949385/f07c5882-c528-4b92-bbb3-23918d217746)

Reply:
Agreed, makes sense.